### PR TITLE
Always finish tagged aka. 'important' builds

### PR DESCRIPTION
**Pull request by okurz**
All tests in a build that has been tagged are not obsoleted if a new build
is triggered. Actually any tag is regarded as 'important' but if it has been
tagged there is not much lost if we finish this build. The obsoletion feature
is especially important for fast paced re-scheduling when no one cares to
put a comment anywhere.

As a tag comment can be created at any time regardless if the corresponding
build exists or not builds to come can be marked to be important already
upfront.

Related progress issue: https://progress.opensuse.org/issues/11072

### CODE REVIEW

**Comment by okurz** [comment]
I would like to test this some more but I would be very open for feedback on this. Especially the double-while-for-looping in the second commit I would like to improve if someone has a nice idea.

File: lib/OpenQA/Scheduler/Scheduler.pm [hunk 1]
```diff
@@ -793,8 +793,28 @@ sub job_cancel($;$) {
...
+        # ... or belong to a tagged build, i.e. is considered important
+        # this might be even the tag 'not important' but not much is lost if
+        # we still not cancel these builds
+        my $groups_query = $scheduled_jobs->get_column('group_id')->as_query;
+        my $groups       = schema->resultset("JobGroups")->search($groups_query);
+        my @important_builds;
+        while (my $g = $groups->next) {
+            for my $comment ($g->comments->all) {
+                my @tag = $comment->tag;
+                push @important_builds, $tag[0];
```
Comment (coolo) [inline lib/OpenQA/Scheduler/Scheduler.pm]: It's very early and I only had s small coffee yet, but to me these 7 lines look to me like doing the same as these 2:
```
my $comments = schema->resultset("Comments")->search({group_id => -in => $groups_query});
my @important_builds = map { $_->tag } $comments->all;
```

Comment (okurz) [inline lib/OpenQA/Scheduler/Scheduler.pm]: more or less, yes, see update

**Comment by coolo** [comment]
would work for me

**Comment by okurz** [comment]
Extended test, was missing the test case for "new build after important is not important anymore". Verified manually on my computer by scheduling new isos, commenting on jobs to make them important, commenting on group overview to mark builds as important. Important jobs and builds stay around as scheduled, other jobs and builds are cancelled as in before. There might be unforeseen consequences on chained jobs in case our test coverage is incomplete or everything might just turn out to be ok. So, go ahead, merge :-)

**Comment by aaannz** [comment]
:+1:

**Pull request merged**